lib/renderer/src/renderer/scene/rust/vange_rs.h

```diff
@@ -0,0 +1,82 @@
+#ifndef VANGE_RS_H
+#define VANGE_RS_H
+
+#include <cstdint>
+
+/**
+  This is a stub file for the upcoming vange-rs library bindings
+**/
+
+// TODO: agree on how we deal with structs
+struct vange_rs_camera_description {
+	float fov;
+	float aspect;
+	float near;
+	float far;
+};
+
+struct vange_rs_vector3 {
+	float x;
+	float y;
+	float z;
+};
+
+struct vange_rs_quaternion {
+	float x;
+	float y;
+	float z;
+	float w;
+};
+
+struct vange_rs_transform {
+	vange_rs_vector3 position;
+	vange_rs_quaternion rotation;
+};
+
+struct vange_rs_map_description {
+	int32_t width; // H_SIZE
+	int32_t height; // V_SIZE
+
+	// data layout: first `width` bytes are height, the rest `width` bytes are meta data
+	uint8_t** lineT;
+
+	// Material offsets in the palette
+	uint8_t* material_begin_offsets;
+	uint8_t* material_end_offsets;
+
+	// 8 for world, 16 for escave
+	int32_t material_count;
+};
+
+// TODO: or it could be left, bottom, right, top
+struct vange_rs_rect {
+	int32_t x;
+	int32_t y;
+	int32_t width;
+	int32_t height;
+};
+
+
+// TODO: pointer?
+void vange_rs_camera_create(vange_rs_camera_description desc);

```

Comment:
it would be good to have a context object explicit, i.e.
```cpp
typedef void* rv_context;
rv_context rv_init();
rv_exit(rv_context);
rv_camera_init(rv_context, rv_camera_desc);
```

And so forth, every function having "rv_context" as the first argument.